bofire/benchmarks/aspen_benchmark.py

```diff
@@ -0,0 +1,180 @@
+import logging
+import os
+
+import pandas as pd
+import win32com.client as win32
+
+from bofire.benchmarks.benchmark import Benchmark
+from bofire.domain import Domain
+
+# Create a Logger that safes console output.
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+formatter = logging.Formatter("%(asctime)s:%(levelname)s:%(message)s")
+if not os.path.exists("bofire_logs"):
+    os.makedirs("bofire_logs")
+file_handler = logging.FileHandler("bofire_logs/aspen_benchmark.log")
+file_handler.setFormatter(formatter)
+logger.addHandler(file_handler)
+
+
+class Aspen_benchmark(Benchmark):
+    """This class connects to a Aspen plus file that runs the desired process.
+    It writes incoming input values into Aspen plus, runs the simulation and returns the results.
+    When initializing this class, make sure not to block multiple Aspen plus licenses at once
+    when is not absolutely needed.
+
+    Args:
+        Benchmark: Subclass of the Benchmark function class.
+    """
+
+    def __init__(
+        self, filename: str, domain: Domain, paths: list[list[dict[str, str]]]
+    ) -> None:
+        """Initializes Aspen_benchmark. A class that connects to Aspen plus.
+
+        Args:
+            filename (str): Filepath of the Aspen plus simulation file.
+            domain (Domain): Domain of the benchmark setting inclunding bounds and information about input values.
+            paths (list[list[dict[str, str]]]): 2xn list with dictionaries containing the filepaths to the variables in the Aspen variable tree
+            in Aspen plus {"name": "path"}.
+
+        Raises:
+            SystemExit: In case the number of provided variable names does not match the number of provided Aspen variable tree paths.
+        """
+
+        self.filename = filename
+        self._domain = domain
+        self.paths = paths
+
+        self.aspen_is_running = False
+
+        # Get the variable names (keys) from the domain to access them later easily.
+        self.keys = [[], []]  # keys[0] for x, keys[1] for y
+        for feature in self.domain.inputs.features:
+            self.keys[0].append(feature.key)
+        for feature in self.domain.outputs.features:
+            self.keys[1].append(feature.key)
+
+        # Check, if number of paths matches number of variables
+        for key_list, path_list in zip(self.keys, self.paths):
```

Comment:
you should also check that every variable appears in the path dict. Not only the numbers. It could be that the names of the feature keys and names of the paths do not match.